src/metadata/loader.rs

```diff
@@ -627,3 +629,21 @@ fn text<R: BufRead>(reader: &mut Reader<R>, name: &[u8]) -> Result<String, error
 		}
 	}
 }
+
+fn text_check_regex<R: BufRead>(reader: &mut Reader<R>, name: &[u8]) -> Result<String, error::LoadMetadata> {
+	let regex_source = text(reader, name)?;
+	// check regular expression syntax
```

Comment:
Why isn't this `check_regex(text(reader, name)?)`?

Reply:
dunno, just happened like this :-D
also one returns a fresh String, the other works with an existing &str

Reply:
i refactored it so that at least the regex check is not implemented twice...